apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.js

```diff
@@ -25,144 +29,156 @@ function obtainInfo(req, res, next) {
   );
   res.locals.groupsCsvFilename = prefix + 'groups.csv';
 
-  //config and group info
-  const params = { assessment_id: res.locals.assessment.id };
-  sqldb.query(sql.config_info, params, function (err, result) {
-    if (ERR(err, next)) return;
-    res.locals.isGroup = true;
-    if (result.rowCount === 0) {
-      res.locals.isGroup = false;
-      res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
-      return;
-    }
-    res.locals.config_info = result.rows[0];
-    res.locals.config_info.defaultMin = res.locals.config_info.minimum || 2;
-    res.locals.config_info.defaultMax = res.locals.config_info.maximum || 5;
+  const groupConfig = await sqldb.queryOptionalRow(
+    sql.config_info,
+    { assessment_id: res.locals.assessment.id },
+    GroupConfigSchema
+  );
+  res.locals.isGroup = !!groupConfig;
+  if (!groupConfig) {
+    res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+    return;
+  }
+  res.locals.config_info = groupConfig;
+  res.locals.config_info.defaultMin = groupConfig.minimum || 2;
+  res.locals.config_info.defaultMax = groupConfig.maximum || 5;
 
-    const params = {
+  res.locals.assessment_list_rows = await sqldb.queryRows(
+    sql.assessment_list,
+    {
       assessment_id: res.locals.assessment.id,
       course_instance_id: res.locals.config_info.course_instance_id,
+    },
+    z.object({
+      id: IdSchema,
+      tid: z.string().nullable(),
+      title: z.string().nullable(),
+    })
+  );
+
+  res.locals.groups = await sqldb.queryRows(
+    sql.select_group_users,
+    {
       group_config_id: res.locals.config_info.id,
-    };
-    sqldb.query(sql.assessment_list, params, function (err, result) {
-      if (ERR(err, next)) return;
-      res.locals.assessment_list_rows = result.rows;
-      sqldb.query(sql.select_group_users, params, function (err, result) {
-        if (ERR(err, next)) return;
-        res.locals.groups = result.rows;
-        sqldb.query(sql.select_not_in_group, params, function (err, result) {
-          if (ERR(err, next)) return;
-          res.locals.notAssigned = result.rows;
-          debug('render page');
-          res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
-        });
-      });
-    });
-  });
+    },
+    z.object({
+      group_id: IdSchema,
+      name: z.string(),
+      size: z.number(),
+      // TODO: what actually is the type of this thing?
+      uid_list: z.array(z.string()),
```

Comment:
That seems like the correct type to me.

Reply:
Thanks for the reminder to remove this - left over from a WIP!